clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -440,7 +458,8 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                           TRACE_MARKER_TYPE_KERNEL_XFER)) ||
                     // We expect a gap on a window transition.
                     shard->window_transition_ ||
-                    shard->prev_instr_.instr.type == TRACE_TYPE_INSTR_SYSENTER,
+                    shard->prev_instr_.instr.type == TRACE_TYPE_INSTR_SYSENTER ||
+                    shard->saw_double_syscall_instrs_with_same_pc_,
                 "Non-explicit control flow has no marker");
```

Comment:
So we'll report the new invariant error type and the old one too? I think we need only one.

Reply:
I read this wrong. Ignore.

Reply:
`saw_double_syscall_instrs_with_same_pc_` avoids it from the being reported as both types.